src/modules/adaptors/Compound/V3/SupplyAdaptor.sol

```diff
@@ -0,0 +1,177 @@
+        return keccak256(abi.encode("Supply Adaptor V 0.0"));
+    }
+
+    //============================================ Implement Base Functions ===========================================
+    /**
+     * @notice Deposit base token into Compound V3.
+     */
+    function deposit(uint256 assets, bytes memory adaptorData, bytes memory) public override {
+        IComet comet = abi.decode(adaptorData, (IComet));
+
+        _verifyComet(comet);
+
+        ERC20 base = comet.baseToken();
+        base.safeApprove(address(comet), assets);
+
+        comet.supply(address(base), assets);
+
+        _revokeExternalApproval(base, address(comet));
+    }
+
+    /**
+     * @notice Withdraw base token from Compound V3 and send to receiver.
+     */
+    function withdraw(
+        uint256 assets,
+        address receiver,
+        bytes memory adaptorData,
+        bytes memory configurationData
+    ) public override {
+        _externalReceiverCheck(receiver);
+
+        bool isLiquid = abi.decode(configurationData, (bool));
+        if (!isLiquid) revert BaseAdaptor__UserWithdrawsNotAllowed();
+
+        IComet comet = abi.decode(adaptorData, (IComet));
+
+        _verifyComet(comet);
+
+        ERC20 base = comet.baseToken();
+
+        comet.withdrawTo(receiver, address(base), assets);
+    }
+
+    /**
+     * @notice If isLiquid is true, reports the minimum between baseSupplied and baseLiquid.
+     *         else reports 0.
+     */
+    function withdrawableFrom(
+        bytes memory adaptorData,
+        bytes memory configurationData
+    ) public view override returns (uint256) {
+        bool isLiquid = abi.decode(configurationData, (bool));
+        if (isLiquid) {
+            IComet comet = abi.decode(adaptorData, (IComet));
+            uint256 baseSupplied = comet.balanceOf(msg.sender);
+            uint256 baseLiquid = comet.baseToken().balanceOf(address(comet));
+            return baseSupplied > baseLiquid ? baseLiquid : baseSupplied;
+        } else return 0;
+    }
+
+    /**
+     * @notice Returns the balance of comet base token.
+     */
+    function balanceOf(bytes memory adaptorData) public view override returns (uint256) {
+        IComet comet = abi.decode(adaptorData, (IComet));
+        return comet.balanceOf(msg.sender);
+    }
+
+    /**
+     * @notice Returns `comet.baseToken()`
+     */
+    function assetOf(bytes memory adaptorData) public view override returns (ERC20) {
+        IComet comet = abi.decode(adaptorData, (IComet));
+        return comet.baseToken();
+    }
+
+    /**
+     * @notice This adaptor returns collateral, and not debt.
+     */
+    function isDebt() public pure override returns (bool) {
+        return false;
+    }
+
+    //============================================ Strategist Functions ===========================================
+
+    /**
+     * @notice Allows strategist to supply base token to Compound V3.
+     */
+    function supplyBase(IComet comet, uint256 assets) external {
```

Comment:
It is weird behavior, but I havent thought of any issues with them doing this. Yes it is weird, but its not like they are skirting any HF checks, or taking on a position they shouldnt be. The cellar must be setup to use the BorrowAdaptor in order to take on debt, so if it also has access to the supply adaptor, then I dont see any issues with a strategist doing this, though it would be less convenient since it does not cap the repay amount at the actual debt owed.

My thinking right now is that this is okay to leave this.

Reply:
Perhaps we should leave this comment open and get the external auditor's point of view on it as well. I agree with your conclusion though.